cocos/renderer/backend/ProgramState.cpp

```diff
@@ -262,22 +264,44 @@ ProgramState *ProgramState::clone() const
 
 void ProgramState::createVertexUniformBuffer()
 {
+    uint32_t totalUniformsSize = 0;
     const auto& vertexUniformInfos = _program->getVertexUniformInfos();
     for(const auto& uniformInfo : vertexUniformInfos)
     {
         if(uniformInfo.second.bufferSize)
+        {
             _vertexUniformInfos[uniformInfo.second.location] = uniformInfo.second; 
+            totalUniformsSize += uniformInfo.second.bufferSize;
+        }
+    }
+#ifdef CC_USE_METAL
+    if (totalUniformsSize > 0)
+    {
+        _vertexUniformBuffer.resize(totalUniformsSize);
+        std::fill(_vertexUniformBuffer.begin(), _vertexUniformBuffer.end(), 0);
     }
+#endif
 }
 
 void ProgramState::createFragmentUniformBuffer()
 {
+    uint32_t totalUniformsSize = 0;
     const auto& fragmentUniformInfos = _program->getFragmentUniformInfos();
     for(const auto& uniformInfo : fragmentUniformInfos)
     {
         if(uniformInfo.second.bufferSize)
+        {
             _fragmentUniformInfos[uniformInfo.second.location] = uniformInfo.second;
+            totalUniformsSize += uniformInfo.second.bufferSize;
+        }
+    }
+#ifdef CC_USE_METAL
+    if (totalUniformsSize > 0)
+    {
+        _fragmentUniformBuffer.resize(totalUniformsSize);
+        std::fill(_fragmentUniformBuffer.begin(), _fragmentUniformBuffer.end(), 0);
```

Comment:
Can just use _fragmentUniformBuffer.resize(totalUniformSize, 0). But i think the default value is 0 for char, so it is not need to initialize it.